src/util/threadpool.rs

```diff
@@ -569,4 +667,109 @@ mod test {
         assert_eq!(task.gid, group2);
         queue.on_task_started(&group2);
     }
+
+    #[test]
+    fn test_small_group_first_queue() {
+        let concurrency_limit = 2;
+        let small_group_tasks_limit = 2;
+        let mut queue = SmallGroupFirstQueue::new(concurrency_limit, small_group_tasks_limit);
+        let mut id = 1;
+        let mut expect_task_id = 0;
+
+        // high_priority_queue and low_priority_queue is both empty.
+        assert!(queue.pop().is_none());
+
+        // Push 2 tasks of `group1` into queue
+        let group1 = 1001;
+        for _ in 0..small_group_tasks_limit {
+            let mut task = Task::new(group1, move || {});
+            task.id = id;
+            id += 1;
+            queue.push(task);
+        }
+
+        // high:[g11,g12]; low:[]
+        assert!(queue.low_priority_queue.is_empty());
+
+        // low_priority_queue is empty
+        let task = queue.pop().unwrap();
+        assert!(task.id > expect_task_id);
+        expect_task_id = task.id;
+        queue.on_task_started(&task.gid);
+
+        // high:[g12]; low:[];running:g11
+        for _ in 0..small_group_tasks_limit {
+            let mut task = Task::new(group1, move || {});
+            task.id = id;
+            id += 1;
+            queue.push(task);
+        }
+
+        // since g11(front in high_queue) comes before g13(front in low_priority_queue),
```

Comment:
Make it constant to use `high/low_queue` or `high/low_priority_queue` in all comments.